### DISCUSSION TITLE
Scaling by 1/a missing in double layer expression?

### DISCUSSION THREAD

**Discussion by bardsleypt** [pybamm-team/PyBaMM#2142] [Q&A] [answered]
I am currently working with simulations that involve electrical double layer capacitance and have come across a scaling (actually lack of a scaling) that might be a bug? I'll give a full derivation of my understanding below, but the short summary is it seems in the nondimensionalization of the non-Faradaic current equation a scaling by 1/a is missing here:

https://github.com/pybamm-team/PyBaMM/blob/e5714a9d747ea772685161043d37d5c99972da99/pybamm/models/submodels/electrolyte_conductivity/surface_potential_form/full_surface_form_conductivity.py#L286

If this is not the case, could someone possibly shed some light on where I'm going wrong? If my understanding is wrong I suspect it is somewhere in the dimensional scalings within the codebase.

I've attached a pdf that shows my derivation of this particular submodel, as well as the dimensional scalings I have found throughout the codebase. I tried to be brief, but its a bit involved so I wanted to give sufficient details. Please let me know if you need any clarifications on any points.

Thanks in advance for any advice.

Best,
Patrick
[dl.pdf](https://github.com/pybamm-team/PyBaMM/files/9031391/dl.pdf)

**Comment by valentinsulzer** [comment]
You're probably right. Would be interesting to see whether this makes any difference to relevant data (pulsing or EIS). 

Good to have you looking over the equations so carefully! Sorry about the slow reply on your SEI question, we're looking into it

**Reply by bardsleypt** [reply]
I don't have any relevant physical data to compare with, but I should be able to generate some synthetic measurements with and without the 1/a scaling just to see how dramatic (if at all) the effect is.

And yes of course, I'm happy to be looking through them. PyBaMM is a great tool and has helped me a lot, hopefully I can help improve it some too. No worries about the pace on SEI growth, I understand it takes time.

**Reply by valentinsulzer** [reply]
Btw, as you say in the pdf, this is dimensionless `a`, so in most cases it will be equal to one anyway. The dimensional factor of `a` is accounted for in the dimensionless scaling already. The only case where `a` changes is if there is loss of active material, which causes `a = 3 eps / R` to change

**Reply by bardsleypt** [reply]
Yes exactly, I don't think this is an issue outside of some of the more exploratory-type models as opposed to the standard DFN model. I'm not certain, but it seems this could also be an issue when the particle sizes are non-uniform as then `a = 3 eps / R(x)` would not exactly equal `a_typ = Avg_x(3 eps / R(x))`, thus `a` would not be identically 1 throughout the domain.

**Comment by bardsleypt** [comment]
I generated a notebook that compares Nyquist EIS plots for two variants of the double layer model (with and without the 1/a scaling), and for two variants of the particle-radii (constant profile and variable profile). As expected, since for spherical particles a = 3*eps/R, the 1/a scaling in the RHS of the overpotential-DE does change the results slightly. I've attached the notebook to this posting (zip and pdf). Let me know if something is unclear, hopefully it is brief enough to follow though.

Also, I have forked PyBaMM and made the single-line change (scaling by 1/a) discussed above. I need to clean up the commit message and run the tests, but once that is done I'll put a PR in place. Then if we decide this scaling should be fixed it'll be a straightforward merge. I'll reference this thread when I make the PR.

[scaled_dl.pdf](https://github.com/pybamm-team/PyBaMM/files/9115682/scaled_dl.pdf)
[scaled_dl.ipynb.zip](https://github.com/pybamm-team/PyBaMM/files/9115685/scaled_dl.ipynb.zip)

**Comment by bardsleypt** [comment, marked as answer]
PR in draft here: https://github.com/pybamm-team/PyBaMM/pull/2175
